File: src/main/java/com/flexpoker/pushnotificationhandlers/TableUpdatedPushNotificationHandler.java
package com.flexpoker.pushnotificationhandlers;

import javax.inject.Inject;

import org.springframework.messaging.simp.SimpMessageSendingOperations;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Component;

import com.flexpoker.framework.pushnotifier.PushNotificationHandler;
import com.flexpoker.pushnotifications.TableUpdatedPushNotification;
import com.flexpoker.table.query.repository.TableRepository;
import com.flexpoker.util.MessagingConstants;

@Component
public class TableUpdatedPushNotificationHandler implements PushNotificationHandler<TableUpdatedPushNotification> {

    private final SimpMessageSendingOperations messagingTemplate;

    private final TableRepository tableRepository;

    @Inject
    public TableUpdatedPushNotificationHandler(SimpMessageSendingOperations messagingTemplate,
            TableRepository tableRepository) {
        this.messagingTemplate = messagingTemplate;
        this.tableRepository = tableRepository;
    }

    @Async
    @Override
    public void handle(TableUpdatedPushNotification pushNotification) {
        var tableDTO = tableRepository.fetchById(pushNotification.getTableId());
        messagingTemplate.convertAndSend(String.format(MessagingConstants.TABLE_STATUS, pushNotification.getGameId(),
                pushNotification.getTableId()), tableDTO);
    }

}

File: src/main/kotlin/com/flexpoker/pushnotificationhandlers/TableUpdatedPushNotificationHandler.kt
package com.flexpoker.pushnotificationhandlers

import com.flexpoker.framework.pushnotifier.PushNotificationHandler
import com.flexpoker.pushnotifications.TableUpdatedPushNotification
import com.flexpoker.table.query.repository.TableRepository
import com.flexpoker.util.MessagingConstants
import org.springframework.messaging.simp.SimpMessageSendingOperations
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Component
import javax.inject.Inject

@Component
class TableUpdatedPushNotificationHandler @Inject constructor(
    private val messagingTemplate: SimpMessageSendingOperations,
    private val tableRepository: TableRepository
) : PushNotificationHandler<TableUpdatedPushNotification> {

    @Async
    override fun handle(pushNotification: TableUpdatedPushNotification) {
        val tableDTO = tableRepository.fetchById(pushNotification.tableId)
        messagingTemplate.convertAndSend(
            String.format(MessagingConstants.TABLE_STATUS, pushNotification.gameId, pushNotification.tableId),
            tableDTO)
    }

}